frontend/src/app/core/state/hal-resource.ts

```diff
@@ -0,0 +1,19 @@
+export interface HalResourceLink {
+  href:string;
+  title:string;
+}
+
+/*
+ * TODO: This typing is not perfect, since overriding with more specific keys that are optional
+ * e.g. `project?:HalResourceLink;` is not possible. This would result in a possible undefined type.
+ * I'm not sure how to fix this, but it works for now.
+ */
+export type HalResourceLinks = Record<string, HalResourceLink|HalResourceLink[]>;
+
+export type CustomTextFormat = 'markdown'|'custom';
+
+export interface CustomText {
```

Comment:
Is this the normal naming for these types of text?

Reply:
The API name for these objects is `Formattable`